File: 2697-lexicographically-smallest-palindrome/2697-lexicographically-smallest-palindrome.js
/**
 * @param {string} s
 * @return {string}
 */
var makeSmallestPalindrome = function(s) {
    let n = s.length;

    let ml = Math.floor(n / 2) - 1;
    let mr = ml + 1;
    if (n % 2 == 1) ml++;

    let ans = "";
    for(let l = ml, r = mr; l >= 0, r < n; l--, r++) {
        if (s[l] != s[r]) {
            if (s.charCodeAt(l) < s.charCodeAt(r)) {
                ans = s[l] + ans + s[l];
            } else {
                ans = s[r] + ans + s[r];
            }
        } else {
            if (l == r) {
                ans = ans + s[l];
            } else {
                ans = s[l] + ans + s[l];
            }                        
        }
    }

    return ans;
};

File: 2697-lexicographically-smallest-palindrome/2697-lexicographically-smallest-palindrome.cpp
class Solution {
public:
    string makeSmallestPalindrome(string s) {
        int n = s.size();
        for (int i = 0; i < n; ++i)
            s[i] = s[n - 1 - i] = min(s[i], s[n - i - 1]);
        return s;
    }
};